src/vector/transaction.rs

```diff
@@ -177,11 +177,54 @@ impl Dataset {
         }
         Ok(Transaction::new(self))
     }
+
+    /// Optionally start a transaction before running `func` for performance
+    ///
+    /// This uses transaction if the dataset supports it, otherwise it
+    /// runs the `func` function as it is on the dataset. If the
+    /// closure results in error, and it is a transaction, it is
+    /// rolled back.
+    pub fn maybe_batch(&mut self, func: impl Fn(&Dataset) -> Result<()>) -> Result<()> {
+        let force = 0; // since this is for speed
+        let rv = unsafe { gdal_sys::GDALDatasetStartTransaction(self.c_dataset(), force) };
+        if rv == OGRErr::OGRERR_UNSUPPORTED_OPERATION {
+            func(self)
+        } else if rv == OGRErr::OGRERR_NONE {
+            let res = func(self);
+            if res.is_ok() {
+                let rv = unsafe { gdal_sys::GDALDatasetCommitTransaction(self.c_dataset()) };
+                if rv != OGRErr::OGRERR_NONE {
+                    Err(GdalError::OgrError {
+                        err: rv,
+                        method_name: "GDALDatasetCommitTransaction",
+                    })
+                } else {
+                    res
+                }
+            } else {
+                let rv = unsafe { gdal_sys::GDALDatasetRollbackTransaction(self.c_dataset()) };
+                if rv != OGRErr::OGRERR_NONE {
+                    Err(GdalError::OgrError {
```

Comment:
Here `res` is `Err`, but we return the rollback error.

Reply:
Should we print something to warn about rollback then? I think it's important to show we couldn't rollback. Especially if it means they can't operate on dataset anymore.

Reply:
Not sure. If you're using this API, I think you know you're eschewing most transactional guarantees, and I'm inlined to say that the first error is the most important one. If your disk is failing or something, that's when it happened.

I'm not sure if I mentioned this, but my use case for this feature is something that processes its inputs and writes a new dataset at the end. If it crashes, I'll just restart it. Rolling back after an error is nice, but I'm never going to look at the output.

Reply:
Ohk, ignored the rollback error. Left a comment there just in case.